font/src/darwin/mod.rs

```diff
@@ -425,9 +425,13 @@ impl Font {
 
         // Strikeout and underline metrics
         // CoreText doesn't provide strikeout so we provide our own
-        let underline_position = self.ct_font.underline_position() as f32;
-        let underline_thickness = self.ct_font.underline_thickness() as f32;
-        let strikeout_position = line_height as f32 / 2. + descent as f32;
+        let underline_position =
+            (self.ct_font.underline_position() - descent)
+            .floor() as f32;
```

Comment:
I think a `.round()` should be more appropriate here. It should also do a `.round()` in the freetype implementation.